src/vm/hints/sha256_utils.rs

```diff
@@ -0,0 +1,266 @@
+    )?;
+
+    let mut padding: Vec<BigInt> = Vec::new();
+
+    for _ in 0..BLOCK_SIZE - 1 {
+        padding.extend(vec![BigInt::zero(); 16]);
+        padding.extend(iv_static.clone());
+        padding.extend(output.clone());
+    }
+
+    vm.segments
+        .write_arg(&mut vm.memory, &sha256_ptr_end, &padding, Some(&vm.prime))
+        .map_err(VirtualMachineError::MemoryError)?;
+    Ok(())
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::types::relocatable::MaybeRelocatable;
+    use crate::utils::test_utils::*;
+    use crate::vm::errors::memory_errors::MemoryError;
+    use crate::vm::hints::execute_hint::BuiltinHintExecutor;
+    use crate::vm::hints::execute_hint::HintReference;
+    use crate::vm::runners::builtin_runner::RangeCheckBuiltinRunner;
+    use crate::vm::vm_memory::memory::Memory;
+    use num_bigint::Sign;
+    static HINT_EXECUTOR: BuiltinHintExecutor = BuiltinHintExecutor {};
+
+    #[test]
+    fn sha256_input_one() {
+        let mut vm = vm_with_range_check!();
+        vm.memory = memory![((0, 1), 7)];
+        vm.run_context.fp = MaybeRelocatable::from((0, 2));
+        let ids = ids!["full_word", "n_bytes"];
+        vm.references = references!(2);
+
+        assert_eq!(
+            sha256_input(&mut vm, &ids, Some(&ApTracking::new())),
+            Ok(())
+        );
+
+        assert_eq!(
+            vm.memory.get(&MaybeRelocatable::from((0, 0))),
+            Ok(Some(&MaybeRelocatable::from(BigInt::one())))
+        );
+    }
+
+    #[test]
+    fn sha256_input_zero() {
+        let mut vm = vm_with_range_check!();
+        vm.memory = memory![((0, 1), 3)];
+        vm.run_context.fp = MaybeRelocatable::from((0, 2));
+        let ids = ids!["full_word", "n_bytes"];
+        vm.references = references!(2);
+
+        assert_eq!(
+            sha256_input(&mut vm, &ids, Some(&ApTracking::new())),
+            Ok(())
+        );
+
+        assert_eq!(
+            vm.memory.get(&MaybeRelocatable::from((0, 0))),
+            Ok(Some(&MaybeRelocatable::from(BigInt::zero())))
+        );
+    }
+
+    #[test]
+    fn sha256_ok() {
+        let mut vm = vm_with_range_check!();
+
+        vm.memory = memory![
+            ((0, 0), (1, 0)),
+            ((0, 1), (2, 0)),
+            ((1, 0), 22),
+            ((1, 1), 22),
+            ((1, 2), 22),
+            ((1, 3), 22),
+            ((1, 4), 22),
+            ((1, 5), 22),
+            ((1, 6), 22),
+            ((1, 7), 22),
+            ((1, 8), 22),
+            ((1, 9), 22),
+            ((1, 10), 22),
+            ((1, 11), 22),
+            ((1, 12), 22),
+            ((1, 13), 22),
+            ((1, 14), 22),
+            ((1, 15), 22),
+            ((2, 9), 0)
+        ];
+        vm.run_context.fp = MaybeRelocatable::from((0, 2));
+        vm.references = references!(2);
+
+        let ids = ids!["sha256_start", "output"];
+
+        assert_eq!(sha256_main(&mut vm, &ids, Some(&ApTracking::new())), Ok(()));
+
+        assert_eq!(
```

Comment:
same comment for these

Reply:
Resolved: https://github.com/lambdaclass/cleopatra_cairo/pull/357/commits/6628578c9ca647bfdc85bf5b4494e2e130911833